application/shared-kernel/ApiCore/Middleware/WebAppMiddleware.cs

```diff
@@ -144,7 +144,18 @@ public static class WebAppMiddlewareExtensions
 
         if (!File.Exists(templateFilePath))
         {
-            throw new FileNotFoundException("index.html does not exist.", templateFilePath);
+            if (Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == "development")
+            {
+                while (!File.Exists(templateFilePath))
+                {
+                    Console.WriteLine("Waiting for index.html to be built...");
+                    Thread.Sleep(1000);
+                }
+            }
+            else
+            {
+                throw new FileNotFoundException("index.html does not exist.", templateFilePath);
+            }
         }
 
```

Comment:
When does this happen? I've not been able to hit this :)

Anyways. I suggest this code to flatten the nesting, and avoid the proces from hanging.

```suggestion
        if (Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == "development")
        {
            var timeout = DateTime.Now.AddSeconds(5);
            while (!File.Exists(templateFilePath) && DateTime.Now < timeout)
            {
                Console.WriteLine("Waiting for index.html to be built...");
                Thread.Sleep(TimeSpan.FromSeconds(1));
            }
        }

        if (!File.Exists(templateFilePath))
        {
            throw new FileNotFoundException("index.html does not exist.", templateFilePath);
        }
```

Just a heads up: In C#, it's generally preferred to use structured logging methods like `LogDebug`, `LogInfo`, `LogWarning`,`LogError`, etc. for their enhanced filtering and structure capabilities. However, in a static class, injecting a logger isn't straightforward. Given that this is solely for development purposes, using `Console.WriteLine` is probably acceptable.

Reply:
It happens now and then when starting up the application - Api builds a prod version of the app, then the development build kicks in (might wiping the dist folder before building)
That said we could try not throwing the error in development mode - can Look into that.